### PR TITLE
Auth: add application credential support

### LINKED ISSUE #73: Add application credentials auth support
The application credential support was already added into the gophercloud:

https://github.com/gophercloud/gophercloud/pull/1224

In addition we have to keep in mind that scope could not be used for app creds: https://github.com/openstack/keystone/blob/462bb4dcb6ec51a3ca6fc38098cb6ee56557529d/keystone/auth/core.py#L271..L272

clouds.yaml ref: https://docs.openstack.org/keystoneauth/latest/plugin-options.html#v3applicationcredential

Required by: https://github.com/terraform-providers/terraform-provider-openstack/issues/637
Relates to: https://github.com/gophercloud/gophercloud/issues/1365

### PR DESCRIPTION
**Pull request by kayrus**
Resolves #73

### CODE REVIEW

**Comment by jtopjian** [comment]
@kayrus Thank you! I did a brief read and it looks good. But can you add some unit tests for this? The `clientconfig` package has a large set of unit tests to make sure the various combinations of authentication result in the right structures.

**Comment by kayrus** [comment]
Is it enough or do I need to add all 4 cases?

File: openstack/clientconfig/requests.go [hunk 1]
```diff
@@ -517,41 +522,65 @@ func v3auth(cloud *Cloud, opts *ClientOpts) (*gophercloud.AuthOptions, error) {
...
-		}
-	} else {
-		// If Domain* is set, but UserDomain* or ProjectDomain* aren't,
-		// then use Domain* as the default setting.
-		cloud = setDomainIfNeeded(cloud)
-
-		if cloud.AuthInfo.ProjectID != "" {
-			scope.ProjectID = cloud.AuthInfo.ProjectID
+	// Application credentials don't support scope
+	if cloud.AuthInfo.ApplicationCredentialID == "" && cloud.AuthInfo.ApplicationCredentialName == "" && cloud.AuthInfo.ApplicationCredentialSecret == "" {
```
Comment (jtopjian) [inline openstack/clientconfig/requests.go]: Let's wrap this up into something like how `isProjectScoped` works:

```go
if !isApplicationCredential(cloud.AuthInfo) {
```

Comment (kayrus) [reply openstack/clientconfig/requests.go]: Done.

**Pull request merged**